### PR TITLE
[#21709] fix: big number issue in swap max value

### LINKED ISSUE #21709: Incorrect value is pasted when "max value" label is tapped on swap
### Problem:
This issue is easily reproducible in case 18 decimal assets are present

### Steps:
1. Go to swap
2. Set up asset with 18 decimals
3. Tap "max value"
4. Observe pasted value

### Actual result:
the 0.41601235629850597 dai is pasted, however, user has 0.416012356298505966 which leads to "something went wrong"

https://github.com/user-attachments/assets/63bc7aaa-a80a-458a-9eb1-93de561480a2

actual user balance 
Image

### Expected result:
The exact value is entered

### OS: 
IOS, Android

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

### PR DESCRIPTION
**Pull request by mohsen-ghafouri**
fixes #21709

### Summary

the 0.41601235629850597 dai is pasted, however, user has 0.416012356298505966 which leads to "something went wrong"

#### Areas that maybe impacted

- swap flow

### Steps to test

1. Go to swap
2. Set up asset with 18 decimals
3. Tap "max value"
4. Observe pasted value

### Result

https://github.com/user-attachments/assets/e1355733-de6f-4063-9b95-f237243d90c9

status: ready

### CODE REVIEW

**Review by smohamedjavid — APPROVED** [review]

File: src/utils/number.cljs [hunk 1]
```diff
@@ -64,10 +64,13 @@
   Example usage:
   (convert-to-whole-number 12345 2) ; => 123.45"
   [amount decimals]
-  (-> amount
-      (/ (Math/pow 10 decimals))
-      (.toFixed decimals)
-      remove-trailing-zeroes))
+  (let [amount-bn  (money/->bignumber amount)
```
Comment (vkjr) [inline src/utils/number.cljs]: Here you can benefit from `->bignumbers` function defined in `utils.money` it give a convenient way to use `when-let` along with conversion multiple parameters to `bignumber`

**Review by vkjr — APPROVED** [review]

File: src/utils/number.cljs (hunk 1 above)
Comment (mohsen-ghafouri) [reply src/utils/number.cljs]: Thanks @vkjr, applied and looks much cleaner now

**Comment by VolodLytvynenko** [comment]
@mohsen-ghafouri Thank you for PR. Take a look found issue

## ISSUE 1: "Undefined is Not a Function" Error After Long Tapping Asset and Selecting Swap Option in Drawer
### Steps:
1. Go to the Wallet page.
2. Long tap any asset to bring up the drawer menu.
3. Select the Swap option from the drawer.

### Actual Result:
An error message "Undefined is not a function" is displayed.

https://github.com/user-attachments/assets/a34fb5bb-3979-4675-9608-258107683972

### Expected Result:
The user should be navigated to the Swap page with the selected asset pre-filled as the asset to pay.

### Devices:
Pixel 7a, Android 13
iPhone 11 Pro Max, iOS 17

### Logs:
[logs (3).zip](https://github.com/user-attachments/files/18024026/logs.3.zip)

**Comment by VolodLytvynenko** [comment]
Unfortunately, this [issue](https://github.com/status-im/status-mobile/pull/21752#issuecomment-2520358994) is blocking me from testing the max value pasting functionality. I can't access the swap page and need to stop testing for now.

**Comment by mohsen-ghafouri** [comment]
@VolodLytvynenko is it reproducible on iOS device? because i can't see it in iPhone 11 Pro Max, iOS 17.2 , or it's just android?

**Comment by mohsen-ghafouri** [comment]
Would you mind to check it again? @VolodLytvynenko

**Comment by VolodLytvynenko** [comment]
Hi @mohsen-ghafour

> @VolodLytvynenko is it reproducible on iOS device? because i can't see it in iPhone 11 Pro Max, iOS 17.2 , or it's just android?

It was reproducible on both OSes

> Would you mind to check it again? @VolodLytvynenko

I will recheck soon. Thank you!

**Comment by VolodLytvynenko** [comment]
@mohsen-ghafouri thank you for fixes. PR is ready to be merged

**Pull request merged**